File: 151002/Poluectov/Rv-project/publisher/src/main/java/com/poluectov/rvproject/dto/user/UserRequestTo.java
package com.poluectov.rvproject.dto.user;

import com.poluectov.rvproject.model.IdentifiedEntity;
import jakarta.validation.constraints.Size;
import lombok.*;
import lombok.experimental.SuperBuilder;
import org.springframework.beans.factory.annotation.Value;

import java.math.BigInteger;


@EqualsAndHashCode(callSuper = true)
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserRequestTo extends IdentifiedEntity {

    private Long id;
    @Size(min = 2, max = 64)
    String login;
    @Size(min = 8, max = 128)
    String password;
    @Size(min = 2, max = 64)
    String firstname;
    @Size(min = 2, max = 64)
    String lastname;
}

File: 151003/Klimovich/Publisher/Models/DTO/RequestTo/UserRequestTo.cs
namespace Publisher.Models.DTO.ResponseTo
{
    public class UserRequestTo
    {
        public int? id { get; set; }
        public string? login { get; set; } = null;
        public string? password { get; set; } = null;
        public string? firstname { get; set; } = null;
        public string? lastname { get; set; } = null;
        public DateTime? modified { get; set; } = null;
    }
}
